theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -63,3 +106,309 @@ def repoclosures(repo, releases, version) {
 
     return closures
 }
+
+def debian_package_version(changelog_path) {
+    return sh(returnStdout: true, script: "head -n1 ${changelog_path} |awk '{print \$2}'|sed 's/(//;s/)//'|cut -f1 -d-|cut -d: -f2").trim()
+}
+
+def build_deb_package_steps(packages_to_build, version, repoowner = 'theforeman', pull_request = false) {
+    def steps = [
+        setup_sources: [:],
+        executes: [:],
+        rsync_packages: [:]
+    ]
+
+    def repos = [:]
+
+    for(int i = 0; i < packages_to_build.size(); i++) {
+        def pkg_to_build = packages_to_build[i]
+        def project = pkg_to_build['name']
+
+        echo "${pkg_to_build}"
+
+        for(int j = 0; j < pkg_to_build['operating_systems'].size(); j++) {
```

Comment:
looking at `find_changed_debs`, I don't see how `operating_systems` can be longer than 1, as you add a new entry for every changed changelog, instead of updating a possibly existing one.

Reply:
Good catch, that logic changed so many times I missed this simplification